From ec6cd9e1063130ed0f28e13cef4fae87a89d788c Mon Sep 17 00:00:00 2001
From: David Faure <faure@kde.org>
Date: Sun, 5 Oct 2025 12:27:29 +0200
Subject: [PATCH 2/3] Fix build with boost 1.89

boost_system has become a headers-only library, there is nothing to link
to and nothing to find_package against (e.g. on OpenSUSE Tumbleweed
its headers are provided by libboost_headers-devel).
https://github.com/boostorg/system/commit/7a495bb46d7ccd808e4be2a6589260839b0fd3a3
https://www.reddit.com/r/openSUSE/comments/1nwy4cw/libboost_system_missing_from_repos/

src/analyze wasn't linking against Boost_SYSTEM_LIBRARY so there
was no point in locating boost_system there.

(cherry picked from commit 99348321819fe8efb3771b2dcd9aaffbc598b271)
---
 CMakeLists.txt             | 8 +++++++-
 src/analyze/CMakeLists.txt | 2 +-
 2 files changed, 8 insertions(+), 2 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index a3dedd0..b60772f 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -32,7 +32,13 @@ if (APPIMAGE_BUILD)
 endif()
 
 include(FeatureSummary)
-find_package(Boost 1.60.0 ${REQUIRED_IN_APPIMAGE} COMPONENTS system filesystem iostreams container)
+find_package(Boost 1.60.0 ${REQUIRED_IN_APPIMAGE} COMPONENTS filesystem iostreams container)
+# boost_system no longer exists in 1.89
+if (Boost_VERSION VERSION_LESS 1.89.0)
+    find_package(Boost ${REQUIRED_IN_APPIMAGE} COMPONENTS system)
+else()
+    set(Boost_SYSTEM_FOUND TRUE)
+endif()
 set_package_properties(Boost PROPERTIES TYPE RECOMMENDED PURPOSE "Boost container libraries can greatly improve performance (via pmr allocators)")
 find_package(Threads REQUIRED)
 find_package(ZLIB REQUIRED)
diff --git a/src/analyze/CMakeLists.txt b/src/analyze/CMakeLists.txt
index ac79961..2640f4c 100644
--- a/src/analyze/CMakeLists.txt
+++ b/src/analyze/CMakeLists.txt
@@ -2,7 +2,7 @@ if (ECM_FOUND)
     include(ECMEnableSanitizers)
 endif()
 
-find_package(Boost 1.41.0 REQUIRED COMPONENTS iostreams program_options system filesystem)
+find_package(Boost 1.41.0 REQUIRED COMPONENTS iostreams program_options filesystem)
 
 configure_file(analyze_config.h.cmake ${CMAKE_CURRENT_BINARY_DIR}/analyze_config.h)
 
-- 
2.55.0

